packages/medusa/src/api-v2/hooks/payment/[provider]/route.ts

```diff
@@ -0,0 +1,25 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { MedusaRequest, MedusaResponse } from "../../../../types/routing"
+
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  try {
+    const { provider } = req.params
+
+    const options = req.scope.resolve(ModuleRegistrationName.PAYMENT).options
+
+    const event = { provider, data: req.body }
+
+    const eventBus = req.scope.resolve("eventBusService")
+
+    // we delay the processing of the event to avoid a conflict caused by a race condition
+    await eventBus.emit("payment.webhook_received", event, {
```

Comment:
todo: those event string should probably be in the utils package in an enum or something. To error prone otherwise, wdyt?